mautic/mautic
# Revert "fix: [DPMMA-1652] add tests" and real fix for reserved keyword in alias

This reverts commit e7b4014.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR revert https://github.com/mautic/mautic/pull/12043.  This PR cause another issues like 
https://github.com/mautic/mautic/pull/12211, https://github.com/mautic/mautic/pull/12267 and others not discovered bugs (3th party as well),

In fact, Mautic does not allow creating fields with reserved keywords in alias. It add prefix  https://github.com/mautic/mautic/blob/2929fc4622bfcfb4caa7ce886c5ecdc18e4e536b/app/bundles/CoreBundle/Model/FormModel.php#L438

But the issue could come If you switch from a different database version, where the list of reserved keywords could be different.  In this case, I suggest adding backticks around the column name just for affected columns and before inserting the query.




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create form and  field function. 
3. Go to related form and form results table and rename field alias from f_function to function and column name of table from results table from f_function to function.
4. Try submit form
5. You should see error
6. After PR should work properly
